src/PrestaShopBundle/Form/Admin/Sell/Product/Stock/StockType.php

```diff
@@ -130,17 +130,29 @@ public function configureOptions(OptionsResolver $resolver)
     {
         $resolver
             ->setDefaults([
-                'label' => false,
+                'label' => $this->trans('Stocks', 'Admin.Catalog.Feature'),
                 'required' => false,
                 'virtual_product_file_id' => null,
                 // Suppliers can be removed so there might be extra data during type switching
                 'allow_extra_fields' => true,
             ])
             ->setRequired([
                 'product_id',
+                'product_type',
             ])
             ->setAllowedTypes('product_id', 'int')
+            ->setAllowedTypes('product_type', 'string')
             ->setAllowedTypes('virtual_product_file_id', ['int', 'null'])
+            ->setNormalizer('label', function (OptionsResolver $resolver) {
+                $productType = $resolver->offsetGet('product_type');
+                if ($productType === ProductType::TYPE_VIRTUAL) {
+                    return $this->trans('Virtual product', 'Admin.Catalog.Feature');
+                } elseif ($productType === ProductType::TYPE_PACK) {
+                    return $this->trans('Packs', 'Admin.Catalog.Feature');
```

Comment:
Sorry if my question is stupid but why don't you manage `ProductType::TYPE_COMBINATIONS` here ? :thinking:

Reply:
Not stupid don't worry you can't know it in advance, it's just because this tab is actually removed when we are displaying a form for combinations This is handled by the ProductTypeListener So since the tab is removed no need to adapt its label for combinations

Reply:
:+1: :student: